log-window.js

```diff
@@ -13,98 +13,252 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  *
- * @fileoverview Creates and communicates with the actual log window.
+ * @fileoverview This is the logic for the actual log window.
```

Comment:
@gvking, I think this being a class will make the stateful formatting more straightforward.  Hope this helps!